### PR TITLE
New flow to narrow traceback duration

### PR DESCRIPTION
**Pull request by smcmurtry**
# Summary | Résumé

Working on #1096. This will remain a draft until the new flow is implemented and mergeable.

# To Do
- [x] Style the picker screen on iOS
- [x] Style the select box to look like a dropdown on Android and iOS
- [x] Add in back button to each page
- [x] Pass the selected date + date type to the backend
- [x] Merge #1094 into this branch
- [x] Make the DataSharing homescreen link to the correct page (could be either `TekUploadSubsequentDays` or `Step2` depending on whether the user has done their initial TEK upload or not
- [x] OTK form page should just have a next button if the user is already diagnosed
- [x] Add in new content to each screen
- [x] Add in "Step x of 4" to each page
- [x] Add radio buttons
- [x] Step 2 - disable next button until a radio item is selected

### CODE REVIEW

**Comment by smcmurtry** [comment]
Here are some screenshots of the new flow: 
Screen Shot 2020-10-05 at 3 55 30 PM

This feature is complete except for testing and more review internally.

File: src/services/BackendService/BackendService.ts [hunk 1]
```diff
@@ -126,22 +131,72 @@ export class BackendService implements BackendInterface {
...
+  saveLastUploadedTekStartTime = async (uploadedTEKs: TemporaryExposureKey[]) => {
+    if (uploadedTEKs.length === 0) {
+      return;
+    }
+    const lastUploadedTekStartTime = uploadedTEKs[0].rollingStartIntervalNumber.toString();
+    await AsyncStorage.setItem(LAST_UPLOADED_TEK_START_TIME, lastUploadedTekStartTime);
+  };
+
+  filterOldTEKs = async () => {
+    const lastUploadedTekStartTime = Number(await AsyncStorage.getItem(LAST_UPLOADED_TEK_START_TIME));
```
Comment (timarney) [inline src/services/BackendService/BackendService.ts:172]: @smcmurtry maybe add some logging here so we know what came out of here + what part of the conditional it's hitting.

File: src/screens/datasharing/SymptomOnsetDateScreen.tsx [hunk 2]
```diff
@@ -1,28 +1,38 @@
...
 
 export const SymptomOnsetDateScreen = () => {
   const i18n = useI18n();
-  const [selectedDate, setSelectedDate] = useState('');
-
+  const navigation = useNavigation();
+  const secondaryButtonOnPress = useCallback(() => navigation.navigate('TekUploadNoDate'), [navigation]);
+  const {data, setSymptomOnsetDate} = useContext(FormContext);
+  const dateParts = data.symptomOnsetDate.split('-');
+  const providedDate = new Date(Number(dateParts[0]), Number(dateParts[1]) - 1, Number(dateParts[2]));
```
Comment (timarney) [inline src/screens/datasharing/SymptomOnsetDateScreen.tsx:17]: @smcmurtry  (provided date) thinking we should break this out - and write a unit test.

File: src/screens/datasharing/TestDateScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+import {BaseTekUploadView, DatePicker, StepXofY} from './components';
+
+export const TestDateScreen = () => {
+  const i18n = useI18n();
+  const {data, setTestDate} = useContext(FormContext);
+  const navigation = useNavigation();
+  const secondaryButtonOnPress = useCallback(() => navigation.navigate('TekUploadNoDate'), [navigation]);
+  const dateParts = data.testDate.split('-');
+  const providedDate = new Date(Number(dateParts[0]), Number(dateParts[1]) - 1, Number(dateParts[2]));
```
Comment (timarney) [inline src/screens/datasharing/TestDateScreen.tsx:17]: As per other comment - could be re-used here

File: src/screens/datasharing/TestDateScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      secondaryButtonOnPress={secondaryButtonOnPress}
+    >
+      <ScrollView style={styles.flex}>
+        <Box paddingHorizontal="m">
+          <StepXofY currentStep={3} />
+          <Text variant="bodyTitle" marginBottom="l" accessibilityRole="header" accessibilityAutoFocus>
+            {i18n.translate('DataUpload.TestDate.Title1')}
+          </Text>
+          <Text marginBottom="l">{i18n.translate('DataUpload.TestDate.Body1')}</Text>
+          <DatePicker daysBack={14} setDate={setTestDate} selectedDate={data.testDate} />
```
Comment (timarney) [inline src/screens/datasharing/TestDateScreen.tsx:32]: For the 14 here - can we pull that as a variable - we be easy to miss this.

File: src/shared/date-fns.ts [hunk 5]
```diff
@@ -11,6 +14,7 @@ export function periodSinceEpoch(date: Date, hoursPerPeriod: number) {
 }
 
 export function daysFromNow(date: Date) {
+  // todo: remove when LastCheckedDisplay is removed
```
Comment (timarney) [inline src/shared/date-fns.ts:17]: PR - here https://github.com/cds-snc/covid-alert-app/pull/1218

**Review by timarney — APPROVED** [review]

**Review by timarney — APPROVED** [review]

**Pull request merged**